diffrax/_solver/kl.py

```diff
@@ -0,0 +1,291 @@
+    (since the diffusion is assumed to be the same). For example, a type
+    of: `MuliTerm(MultiTerm(ODETerm, _DiffusionTerm), ODETerm)`.
+
+    ??? cite "References"
+
+        See section 5 of:
+
+        ```bibtex
+        @inproceedings{li2020scalable,
+            title={Scalable gradients for stochastic differential equations},
+            author={Li, Xuechen and Wong, Ting-Kam Leonard and Chen, Ricky TQ and Duvenaud, David},
+            booktitle={International Conference on Artificial Intelligence and Statistics},
+            pages={3870--3882},
+            year={2020},
+            organization={PMLR}
+        }
+        ```
+
+        Or section 4.3.2 of:
+
+        ```bibtex
+        @article{kidger2022neural,
+            title={On neural differential equations},
+            author={Kidger, Patrick},
+            journal={arXiv preprint arXiv:2202.02435},
+            year={2022}
+        }
+        ```
+    """  # noqa: E501
+
+    solver: AbstractSolver[_SolverState]
+    linear_solver: lx.AbstractLinearSolver = lx.AutoLinearSolver(well_posed=None)
+
+    def order(self, terms: PyTree[AbstractTerm]) -> Optional[int]:
+        return self.solver.order(terms)
+
+    def strong_order(self, terms: PyTree[AbstractTerm]) -> Optional[RealScalarLike]:
+        return self.solver.strong_order(terms)
+
+    def error_order(self, terms: PyTree[AbstractTerm]) -> Optional[RealScalarLike]:
+        if is_sde(terms):
+            order = self.strong_order(terms)
+        else:
+            order = self.order(terms)
+        return order
+
+    @property
+    def term_structure(self):
+        return self.solver.term_structure
+
+    @property
+    def interpolation_cls(self):  # pyright: ignore
+        return self.solver.interpolation_cls
+
+    def init(
+        self,
+        terms: PyTree[AbstractTerm],
+        t0: RealScalarLike,
+        t1: RealScalarLike,
+        y0: KLState,
+        args: Args,
+    ) -> _SolverState:
+        return self.solver.init(terms, t0, t1, y0, args)
+
+    def step(
+        self,
+        terms: PyTree[AbstractTerm],
+        t0: RealScalarLike,
+        t1: RealScalarLike,
+        y0: Y,
+        args: Args,
+        solver_state: _SolverState,
+        made_jump: BoolScalarLike,
+    ) -> tuple[Y, Optional[Y], DenseInfo, _SolverState, RESULTS]:
+        terms1, terms2 = terms.terms
+        drift_term1 = jtu.tree_map(
+            lambda x: x if isinstance(x, ODETerm) else None,
+            terms1,
+            is_leaf=lambda x: isinstance(x, ODETerm),
+        )
+        drift_term1 = jtu.tree_leaves(
+            drift_term1, is_leaf=lambda x: isinstance(x, ODETerm)
+        )
+        drift_term2 = jtu.tree_map(
+            lambda x: x if isinstance(x, ODETerm) else None,
+            terms2,
+            is_leaf=lambda x: isinstance(x, ODETerm),
+        )
+        drift_term2 = jtu.tree_leaves(
+            drift_term2, is_leaf=lambda x: isinstance(x, ODETerm)
+        )
```

Comment:
I don't think these should be necessary -- we can just assume the appropriate terms have been passed outside of any tree structure. (E.g. see how the other SDE-specific solvers work.)